mttl/models/library/expert_library.py

```diff
@@ -330,32 +353,54 @@ def get_auxiliary_data(
             List[Any]: _description_
         """
         # all auxiliary data should have "bin" extension
-        path = self.snapshot_download(self.repo_id, allow_patterns=f"*.{data_type}.bin")
+        has_auxiliary_data = []
+        for file in self.list_repo_files(self.repo_id):
+            if f"{data_type}.bin" in file:
+                try:
+                    name, data_type, _ = os.path.basename(file).split(".")
+                except:
+                    continue
+                has_auxiliary_data.append(name)
+
+        def download_auxiliary(name):
+            path_or_bytes = self.hf_hub_download(
+                self.repo_id, filename=f"{name}.{data_type}.bin"
+            )
+            return torch.load(path_or_bytes, map_location="cpu", weights_only=False)
+
+        expert_names = [expert_name] if expert_name is not None else self.keys()
+        expert_names_to_process = [
+            name for name in expert_names if name in has_auxiliary_data
+        ]
+
+        # Use ThreadPoolExecutor for multithreading
```

Comment:
Would `snapshot_download` with an `allow_patterns` work here?

Reply:
yeah, no, I didn't implement `allow_patterns` for VirtualLocalLibrary..., also `snapshot_download` is not used anymore, i am not sure if it affects performance

Reply:
Performance should be similiar. I'm asking because snapshot_download already download multiple files in parallel. I'm ok with your approach using threads, feel free to merge the PR!